File: backend/Recintos-Backend/src/main/java/com/recintos/municipalidad/model/Recinto.java
package com.recintos.municipalidad.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "recinto")
public class Recinto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idRecinto;

    private String nombre;
    private String ubicacion;
    private String descripcion;
    private String imagenUrl;
    private Integer capacidad;
    private String coordenadasGPS;
    private String estado;

    @OneToMany(mappedBy = "recinto")
    @com.fasterxml.jackson.annotation.JsonIgnore
    private List<Evento> eventos;
}

File: kotlin-app/MunicipalidadRecintos/composeApp/src/commonMain/kotlin/org/example/project/domain/model/Recinto.kt
package org.example.project.domain.model

data class Recinto(
    val idRecinto: String? = null,
    val nombre: String? = null,
    val ubicacion: String? = null,
    val descripcion: String? = null,
    val imagenUrl: String? = null,
    val coordenadasGPS: String? = null,
    val capacidad: Int? = null,
    val estado: String? = null
)

data class Categoria(
    val id: Int,
    val nombre: String,
    val descripcion: String
)
